zrml/prediction-markets/src/lib.rs

```diff
@@ -902,39 +918,56 @@ mod pallet {
             T::MarketCommons::mutate_market(&market_id, |market| {
                 ensure!(market.report.is_none(), Error::<T>::MarketAlreadyReported);
                 Self::ensure_market_is_closed(market)?;
-                ensure!(
-                    market.matches_outcome_report(&market_report.outcome),
-                    Error::<T>::OutcomeMismatch
-                );
 
                 let mut should_check_origin = false;
                 match market.period {
                     MarketPeriod::Block(ref range) => {
-                        if current_block
-                            <= range.end.saturating_add(T::ReportingPeriod::get().into())
-                        {
+                        let oracle_delay_end =
+                            range.end.saturating_add(market.deadlines.oracle_delay.into());
+                        ensure!(
+                            oracle_delay_end < current_block,
+                            Error::<T>::NotAllowedToReportYet
+                        );
+                        let oracle_duration_end = oracle_delay_end
+                            .saturating_add(market.deadlines.oracle_duration.into());
+                        if oracle_duration_end >= current_block {
```

Comment:
```suggestion
                        if  current_block <= oracle_duration_end {
```

Reply:
not sure why this change suggested? was earlier comment only meant for ensure!() ?

Reply:
I suggest that as a nitpick to have a chronological process.